src/misc/lv_str.c

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file lv_str.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "lv_str.h"
+#include "lv_assert.h"
+#include "lv_log.h"
+
+#include <string.h>
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+/**********************
+ *      MACROS
+ **********************/
+
+/**********************
+ *   GLOBAL FUNCTIONS
+ **********************/
+
+char * lv_strncpy(char * dst, const char * src, size_t count)
+{
+    LV_ASSERT_VERIFY(dst);
+    LV_ASSERT_VERIFY(src);
+
+    char *ret = strncpy(dst, src, count);
```

Comment:
With the current code the string is iterated twice: once in `strncpy` and once in `strlen`. 
What about something like this:
```c
LV_ASSERT_VERIFY(count > 0);
size_t i;
for(i = 0; i < count - 1 && src[i] != '\0'; i++) {
   dst[i] = src[i];
}
dst[i] = '\0';
```

Reply:
With this code the test in ade7126 failed, I had to replace `count -1` with `count`